### PR TITLE
cleanup: Fix a few more clang-tidy warnings.

### PR DESCRIPTION
**Pull request by iphydf**
---

This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2405)

### CODE REVIEW

**Review by Green-Sky — APPROVED** [review]

File: other/bootstrap_daemon/src/config.c [hunk 1]
```diff
@@ -156,7 +156,7 @@ int get_general_config(const char *cfg_file_path, char **pid_file_path, char **k
     }
 
     // Get PID file location
-    const char *tmp_pid_file;
+    const char *tmp_pid_file = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/config.c:159]: `nullptr`? We use `nullptr` elsewhere in this file.

File: other/bootstrap_daemon/src/config.c [hunk 2]
```diff
@@ -169,7 +169,7 @@ int get_general_config(const char *cfg_file_path, char **pid_file_path, char **k
     memcpy(*pid_file_path, tmp_pid_file, pid_file_path_len);
 
     // Get keys file location
-    const char *tmp_keys_file;
+    const char *tmp_keys_file = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/config.c:172]: `nullptr`?

File: other/bootstrap_daemon/src/config.c [hunk 3]
```diff
@@ -228,7 +228,7 @@ int get_general_config(const char *cfg_file_path, char **pid_file_path, char **k
 
     if (*enable_motd) {
         // Get MOTD
-        const char *tmp_motd;
+        const char *tmp_motd = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/config.c:231]: `nullptr`?

File: other/bootstrap_daemon/src/config.c [hunk 4]
```diff
@@ -338,17 +338,17 @@ int bootstrap_from_config(const char *cfg_file_path, DHT *dht, int enable_ipv6)
         return 1;
     }
 
-    int bs_port;
-    const char *bs_address;
-    const char *bs_public_key;
+    int bs_port = 0;
+    const char *bs_address = NULL;
+    const char *bs_public_key = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/config.c:343]: `nullptr`?

File: other/bootstrap_daemon/src/config.c [hunk 5]
```diff
@@ -338,17 +338,17 @@ int bootstrap_from_config(const char *cfg_file_path, DHT *dht, int enable_ipv6)
...
 
-    config_setting_t *node;
+    config_setting_t *node = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/config.c:345]: `nullptr`?

File: other/bootstrap_daemon/src/config.c [hunk 6]
```diff
@@ -338,17 +338,17 @@ int bootstrap_from_config(const char *cfg_file_path, DHT *dht, int enable_ipv6)
...
 
     int i = 0;
 
     while (config_setting_length(node_list)) {
-        int address_resolved;
-        uint8_t *bs_public_key_bin;
+        int address_resolved = 0;
+        uint8_t *bs_public_key_bin = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/config.c:351]: `nullptr`?

File: other/bootstrap_daemon/src/tox-bootstrapd.c [hunk 7]
```diff
@@ -61,7 +61,7 @@ static int manage_keys(DHT *dht, char *keys_file_path)
 {
     enum { KEYS_SIZE = CRYPTO_PUBLIC_KEY_SIZE + CRYPTO_SECRET_KEY_SIZE };
     uint8_t keys[KEYS_SIZE];
-    FILE *keys_file;
+    FILE *keys_file = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/tox-bootstrapd.c:64]: Shouldn't that be `nullptr`? We use `nullptr` elsewhere in this file.

File: other/bootstrap_daemon/src/tox-bootstrapd.c [hunk 8]
```diff
@@ -120,7 +120,7 @@ static void print_public_key(const uint8_t *public_key)
 static void daemonize(LOG_BACKEND log_backend, char *pid_file_path)
 {
     // Check if the PID file exists
-    FILE *pid_file;
+    FILE *pid_file = NULL;
```
Comment (nurupo) [inline other/bootstrap_daemon/src/tox-bootstrapd.c:123]: `nullptr`?

**Review by nurupo — CHANGES_REQUESTED** [review]

**Comment by nurupo** [comment]
>It's unnecessary in any of these cases, but having the warning is
>useful, and may catch errors in the future.

Which warning? Variable assigned but never used?

**Comment by nurupo** [comment]
Speaking of warnings, is there a way to make a custom token warning, so that we would get warned if someone uses `NULL` instead of `nullptr`?

**Comment by iphydf** [comment]
> Speaking of warnings, is there a way to make a custom token warning, so that we would get warned if someone uses `NULL` instead of `nullptr`?

We have that, yes. In one of the builds we get an error for it.

**Comment by iphydf** [comment]
> >It's unnecessary in any of these cases, but having the warning is
> >useful, and may catch errors in the future.
> 
> Which warning? Variable assigned but never used?

"Variable not initialised"

File: other/bootstrap_daemon/src/tox-bootstrapd.c (hunk 8 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/tox-bootstrapd.c]: Done.

File: other/bootstrap_daemon/src/tox-bootstrapd.c (hunk 7 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/tox-bootstrapd.c]: Done.

File: other/bootstrap_daemon/src/config.c (hunk 6 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/config.c]: Done.

File: other/bootstrap_daemon/src/config.c (hunk 5 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/config.c]: Done.

File: other/bootstrap_daemon/src/config.c (hunk 4 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/config.c]: Done.

File: other/bootstrap_daemon/src/config.c (hunk 3 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/config.c]: Done.

File: other/bootstrap_daemon/src/config.c (hunk 2 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/config.c]: Done.

File: other/bootstrap_daemon/src/config.c (hunk 1 above)
Comment (iphydf) [reply other/bootstrap_daemon/src/config.c]: Done.

**Comment by iphydf** [comment]
I've changed it back. This change conflicts too much with other warnings that are more useful.

**Review by JFreegman — APPROVED** [review]
> Reviewed 3 of 87 files at r1, 84 of 84 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2405)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @nurupo)

**Comment by iphydf** [comment]
@nurupo PTAL. I've addressed your comments, I think (and also reverted most of it).

**Comment by nurupo** [comment]
Ah, sorry for the wait, code review replies and @ mentions don't stand out from regular TokTok email spam in my mailbox so I almost missed this, opened it by a pure chance just now.

**Review by nurupo — APPROVED** [review]
LGTM

**Comment by nurupo** [comment]
>code review replies and @ mentions don't stand out from regular TokTok email spam in my mailbox

Actually that's false, they do stand out, was just looking at the wrong gmail label when writing this.

**Pull request merged**